packages/nuxt/src/app/components/client-only.mjs

```diff
@@ -73,3 +94,50 @@ export function createClientOnly (component) {
 
   return clone
 }
+
+function getStaticVNode (vnode) {
+  const fragment = getFragmentHTML(vnode.el)
+
+  if (fragment.length === 0) {
+    return null
+  }
+  return createStaticVNode(fragment.join(''), fragment.length)
+}
```

Comment:
Is there any way you think we can cut down on the extra code here and reduce bundle size when using client components? Is any of this shared with our server-component implementation?

Reply:
sure we can :+1: ! Server components are just a copy/paste of theses lines of code 😃